### PR TITLE
(maint) Turn on string externalization in Facter

### PR DESCRIPTION
**Pull request by Magisus**
This enables string externalization in Facter, and fixes linker errors
on various platforms that were introduced as a result of adding
Leatherman and Boost locales.

The log function used in exe/facter.cc (and its associated tests) does not go
through Leatherman's formatting, and therefore cannot use `{1}`
substitution, but instead must use `%1%`.

### CODE REVIEW

**Comment by Magisus** [comment]
Travis and AppVeyor will pass once cpp-hocon has been tagged with [these](https://github.com/puppetlabs/cpp-hocon/pull/79) updates.

File: lib/CMakeLists.txt [hunk 1]
```diff
@@ -301,10 +301,10 @@ endif()
 
 # Link in additional libraries
```
Comment (MikaelSmith) [inline lib/CMakeLists.txt]: Please add the note about assumptions of boost (static), leatherman (shared).

**Comment by MikaelSmith** [comment]
And update the commit description about the issue in exe/facter.cc with logging.

**Comment by Magisus** [comment]
Going to close and reopen this to reset Travis and Appveyor now that cpp-hocon has been tagged.

**Comment by Magisus** [comment]
Green builds are https://jenkins.puppetlabs.com/view/puppet-agent/view/ad%20hoc/job/platform_puppet-agent_pkg-van-ship_manual-ad-hoc/358/ and https://jenkins.puppetlabs.com/view/puppet-agent/view/ad%20hoc/job/platform_puppet-agent_pkg-van-ship_manual-ad-hoc/359/ (second one is after fixing OSX typo).

**Pull request merged**